src/sif/transitcost.cc

```diff
@@ -568,7 +568,12 @@ Cost TransitCost::EdgeCost(const baldr::DirectedEdge* edge,
                            const baldr::TransitDeparture* departure,
                            const uint32_t curr_time) const {
   // Separate wait time from time on transit
-  float wait_time = departure->departure_time() - curr_time;
+  // adjust the time if it is after midnight.
+  uint32_t current = curr_time;
+  while (current >= 86400)
+    current -= 86400;
+
+  float wait_time = departure->departure_time() - current;
```

Comment:
As far as I can tell if a departure time (seconds) is > 86400 you subtract 86400 in valhalla_build_transit and store that as the departure time. So one minute after midnight has a departure time of 60. If current time is one minute before midnight its value is 86340. So wait_time = 60 - 86340 - leading to a negative wait time.